### ISSUE TITLE
Singularity version on Cartesius cluster @SURFSara gives ValueError

### ISSUE THREAD

**Issue by sverhoeven** [eWaterCycle/grpc4bmi#88]
The version installed is `3.7.1-1.el7_9`. the semver package gives ValueError

**Comment by sverhoeven** [comment]
According to [the spec](https://semver.org/#semantic-versioning-specification-semver) `_` is not allowed.

**Comment by sverhoeven** [comment]
0.2.x version which is being run on Cartesius has been fixed. And PR #90 will fix it in master branch.

**Comment by sverhoeven** [comment]
On snellius apptainer 1.1.5-2.el8 is installed which fails current implementation.

**Comment by SarahAlidoost** [comment]
It seems this is not fixed in `0.2.16` used in [`comparison` env](https://github.com/eWaterCycle/comparison-study/blob/main/environment.yml) , see

```bash
grpc4bmi                  0.2.16                   pypi_0    pypi
```

```python
---------------------------------------------------------------------------
InvalidVersion                            Traceback (most recent call last)
Cell In[14], line 5
      1 model = ewatercycle.models.Lisflood(
      2     version=MODEL_VERSION, parameter_set=parameter_set, forcing=forcing
      3 )
----> 5 config_file, config_dir = model.setup(
      6     MaskMap=MODEL_MASK, cfg_dir=f"{WORK_DIR}/model_output"
      7 )
      9 print(config_file)
     10 print(config_dir)

File /gpfs/home2/fakhereh/mambaforge-pypy3/envs/comparison/lib/python3.10/site-packages/ewatercycle/models/lisflood.py:120, in Lisflood.setup(self, IrrigationEfficiency, start_time, end_time, MaskMap, cfg_dir)
    118 if CFG["container_engine"].lower() == "singularity":
    119     image = get_singularity_image(self.version, CFG["singularity_dir"])
--> 120     self.bmi = BmiClientSingularity(
    121         image=str(image),
    122         input_dirs=input_dirs,
    123         work_dir=str(cfg_dir_as_path),
    124         timeout=300,
    125     )
    126 elif CFG["container_engine"].lower() == "docker":
    127     image = get_docker_image(self.version)

File /gpfs/home2/fakhereh/mambaforge-pypy3/envs/comparison/lib/python3.10/site-packages/grpc4bmi/bmi_client_singularity.py:208, in BmiClientSingularity.__init__(self, image, work_dir, input_dirs, delay, timeout, capture_logs)
    205     msg = f'type of argument "input_dirs" must be collections.abc.Iterable; ' \
    206           f'got {qualified_name(input_dirs)} instead'
    207     raise TypeError(msg)
--> 208 check_singularity_version()
    209 host = 'localhost'
    210 port = BmiClient.get_unique_port(host)

File /gpfs/home2/fakhereh/mambaforge-pypy3/envs/comparison/lib/python3.10/site-packages/grpc4bmi/bmi_client_singularity.py:36, in check_singularity_version()
     34 if p.returncode != 0:
     35     raise SingularityVersionException('Unable to determine singularity version')
---> 36 check_singularity_version_string(stdout.decode('utf-8'))

File /gpfs/home2/fakhereh/mambaforge-pypy3/envs/comparison/lib/python3.10/site-packages/grpc4bmi/bmi_client_singularity.py:21, in check_singularity_version_string(version_output)
     19 def check_singularity_version_string(version_output: str) -> bool:
     20     (app, _, version) = version_output.split(' ')
---> 21     local_version = Version(version)
     22     if app == 'singularity' and local_version not in SpecifierSet(SUPPORTED_SINGULARITY_VERSIONS):
     23         raise SingularityVersionException(f'Unsupported version ({version_output}) of singularity found, '
     24                                           f'supported versions {SUPPORTED_SINGULARITY_VERSIONS}')

File /gpfs/home2/fakhereh/mambaforge-pypy3/envs/comparison/lib/python3.10/site-packages/packaging/version.py:198, in Version.__init__(self, version)
    196 match = self._regex.search(version)
    197 if not match:
--> 198     raise InvalidVersion(f"Invalid version: '{version}'")
    200 # Store the parsed out pieces of the version
    201 self._version = _Version(
    202     epoch=int(match.group("epoch")) if match.group("epoch") else 0,
    203     release=tuple(int(i) for i in match.group("release").split(".")),
   (...)
    209     local=_parse_local_version(match.group("local")),
    210 )

InvalidVersion: Invalid version: '1.1.8-1.el8
'
```
